assets/components/prometheus-operator/manifests/templates/prometheus/ingressperreplica.yaml

```diff
@@ -25,6 +25,7 @@ items:
 {{ toYaml $ingressValues.annotations | indent 8 }}
       {{- end }}
     spec:
+      ingressClassName: contour
```

Comment:
What if someone wants to use different ingress controller with prometheus component? With this patch, this is no longer possible.

Reply:
This PR right now just deals with moving to the apiversion. And not changing any annotations for backwards compatibility reasons they are still valid.

Reply:
Documentation in https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#ingressspec-v1beta1-networking-k8s-io says:

> IngressClassName is the name of the IngressClass cluster resource. The associated IngressClass defines which controller will implement the resource. This replaces the deprecated `kubernetes.io/ingress.class` annotation. For backwards compatibility, when that annotation is set, it must be given precedence over this field. The controller may emit a warning if the field and annotation have different values. Implementations of this API should ignore Ingresses without a class specified. An IngressClass resource may be marked as default, which can be used to set a default value for this field. For more information, refer to the IngressClass documentation.